datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -292,14 +338,53 @@ func datadogSecurityMonitoringRuleSchema() map[string]*schema.Schema {
 
 		"type": {
 			Type:             schema.TypeString,
-			ValidateDiagFunc: validators.ValidateEnumValue(datadogV2.NewSecurityMonitoringRuleTypeReadFromValue),
+			ValidateDiagFunc: validators.ValidateEnumValue(datadogV2.NewSecurityMonitoringRuleTypeReadFromValue, datadogV2.NewSecurityMonitoringSignalRuleTypeFromValue),
 			Optional:         true,
 			Description:      "The rule type.",
 			Default:          "log_detection",
 		},
 	}
 }
 
+// SecurityMonitoringRuleInterface Common Interface to SecurityMonitoringRuleCreateInterface and SecurityMonitoringRuleReadInterface
+type SecurityMonitoringRuleInterface interface {
```

Comment:
I think these interfaces should be private, as they are utilities to factorize code.